packages/core/admin/admin/src/content-manager/components/BlocksEditor/Toolbar/index.js

```diff
@@ -8,27 +8,51 @@ import PropTypes from 'prop-types';
 import { useIntl } from 'react-intl';
 import { Editor, Transforms, Element as SlateElement } from 'slate';
 import { useSlate } from 'slate-react';
-import styled from 'styled-components';
+import styled, { css } from 'styled-components';
 
 import { useBlocksStore } from '../hooks/useBlocksStore';
 import { useModifiersStore } from '../hooks/useModifiersStore';
 
+const ToolbarWrapper = styled(Flex)`
+  ${(props) =>
+    props.disabled &&
+    css`
+      cursor: not-allowed;
+    `}
+`;
+
 const Separator = styled(Toolbar.Separator)`
   background: ${({ theme }) => theme.colors.neutral150};
   width: 1px;
   height: ${pxToRem(24)};
 `;
 
 const FlexButton = styled(Flex).attrs({ as: 'button' })`
-  &:hover {
-    background: ${({ theme }) => theme.colors.primary100};
+  &[aria-disabled] {
+    display: none;
+    color: red;
+    opacity: 0.3;
   }
+  ${(props) =>
+    props.disabled
+      ? css`
+          // Inherit the not-allowed cursor from ToolbarWrapper
+          cursor: inherit;
```

Comment:
no need of just having separate check for cursor, move inside aria-disabled style